TombEngine/Game/effects/tomb4fx.cpp

```diff
@@ -371,31 +371,28 @@ void UpdateFireProgress()
 
 void AddFire(int x, int y, int z, short roomNum, float size, short fade)
 {
-	FIRE_LIST* fptr = &Fires[0];
-	int i = 0;
-	while (fptr->on)
-	{
-		fptr++;
-		if (++i >= MAX_FIRE_LIST)
-			return;
-	}	
+	FIRE_LIST newFire;
 	
 	if (fade)
-		fptr->on = fade;
+		newFire.on = fade;
 	else
-		fptr->on = 1;
+		newFire.on = 1;
+
+	newFire.x = x;
+	newFire.y = y;
+	newFire.z = z;
+	newFire.roomNumber = roomNum;
+	newFire.size = size;
 
-	fptr->x = x;
-	fptr->y = y;
-	fptr->z = z;
-	fptr->roomNumber = roomNum;
-	fptr->size = size;
+	Fires.push_back(newFire);
 }
 
 void ClearFires()
 {
-	for (int i = 0; i < MAX_FIRE_LIST; i++)
-		Fires[i].on = false;
+	for (auto& fire : Fires)
+	{
+		fire.on = false;
```

Comment:
You are not clearing the vector, but instead just setting flag to false. It means that memory leak will occur.

Reply:
ah nice to know. Add the clear method.